src/librustc_lint/builtin.rs

```diff
@@ -637,9 +639,22 @@ impl EarlyLintPass for AnonymousParameters {
                     match arg.pat.node {
                         ast::PatKind::Ident(_, ident, None) => {
                             if ident.name == keywords::Invalid.name() {
-                                cx.span_lint(ANONYMOUS_PARAMETERS,
+                                let ty_snip = cx
+                                    .sess
+                                    .codemap()
+                                    .span_to_snippet(arg.ty.span)
+                                    .unwrap();
```

Comment:
You shouldn't `unwrap()` here.

Instead, in case of a `None`, set it to `"<type>"`.

Also, use `span_suggestion_with_applicability`, with MachineApplicable unless the thing was None, in which case use WithPlaceholders